src/libs/sys-util-rs/LICENSE-APACHE

```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```

Comment:
@jiangliu - ftr, `cargo publish` doesn't work in this crate currently due to a missing version number for the (local) `oci` crate. But that aside, `cargo audit` shows:

```
`license` should be used if the package license can be expressed with a standard SPDX expression.
`license-file` should be used if the package uses a non-standard license.
See https://doc.rust-lang.org/cargo/reference/manifest.html#the-license-and-license-file-fields for more information.
```

As such, I don't see any point in creating the license files for these local crates as the SPDX name is "well-known" (standardised and official). But if you _really want them, why not try specifying something like:

```diff
diff --git a/src/libs/types-rs/Cargo.toml b/src/libs/types-rs/Cargo.toml
index 99eaa8913..ec8322849 100644
--- a/src/libs/types-rs/Cargo.toml
+++ b/src/libs/types-rs/Cargo.toml
@@ -7,13 +7,13 @@ authors = ["The Kata Containers community <kata-dev@lists.katacontainers.io>"]
 repository = "https://github.com/kata-containers/kata-containers.git"
 homepage = "https://katacontainers.io/"
 readme = "README.md"
-license = "Apache-2.0"
+license-file = "../../../LICENSE"
 edition = "2018"

 [dependencies]
 serde = { version = "1.0.100", features = ["derive"] }
 thiserror = "1.0"
```

I haven't verified that `cargo publish` is happy with this, but it looks like it would be given that `cargo package -l --allow-dirty` works as expected.

Reply:
No strong preference here, just because some the project requires license file to be present for each crate. I will remove those license file and add it back if needed when really publishing these crates.